### PR TITLE
feature/1081 add controls completed section

### PR DESCRIPTION
**Pull request by amyd11**
### Overview
Adds controls completed section to the project homepage

### Jira ticket
[ISPGBSS-1081](https://jiraent.cms.gov/browse/ISPGBSS-1081)

### Changes
- added packages:
  - `chart.js` to create infographs
  - `react-chartjs-2` to use the chart.js graphs as react components
- added Controls Completed section to the Project homepage via `ControlsCompletedCTA` component (includes graphic, text, and button to link to project controls page)
- converted control statuses from 'completed' to 'complete' and subsequently some verbiage for these as well
- refactored `Project.jsx` to handle `project` as variable inside component state instead of in context (putting this in context was causing issues when project data was updated and the page was not making a new call to display updated data)
- styling

### Related
- [x] Merging needs coordination with https://github.com/CMSgov/blueprint_api/pull/125

_Screenshot of code running locally_
Screen Shot 2022-09-27 at 3 56 49 PM

### CODE REVIEW

**Review by jwhite3 — APPROVED** [review]

File: src/App.scss [hunk 1]
```diff
@@ -43,6 +43,10 @@ body {
 h1 {
   font-size: 40px;
 }
+.generic-section-header {
+  font-size: 22px;
+  font-weight: bold;
+}
```
Comment (jauschalley) [inline src/App.scss:49]: Praise: A nice little bit of cleanup.

File: src/molecules/ControlsCompletedCTA.jsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  numControlsComplete,
+  numControlsTotal,
+}) {
+  const params = useLocation();
+  const controlsUrl = params.pathname + "/controls";
+  const numControlsIncomplete = numControlsTotal - numControlsComplete;
+  const percentControlsComplete = Math.floor(
+    (numControlsComplete / numControlsTotal) * 100
+  ); // round down to prevent inaccurate "100% complete" feedback as num complete controls near total
+  const data = {
```
Comment (jauschalley) [inline src/molecules/ControlsCompletedCTA.jsx:17]: Nitpik(non-blocking): I know this is exactly how chartjs sets up their examples but a variable name called `data` reminds me of a variable name called `variable` that I saw once.  It feels too generic to me.

File: src/pages/Project.jsx [hunk 3]
```diff
@@ -14,36 +14,35 @@ const ERROR_MESSAGE = "Error loading project";
...
+        setState((state) => ({ ...state, project: response.data })); // TODO: This can be removed or updated accordingly when Breadcrumbs is refactored
+        setProject(response.data);
+        setIsLoading(false);
+      },
+      (err) => {
+        setHasError(true);
+        setIsLoading(false);
+      }
+    );
+  }, [id, setProject, setState]);
```
Comment (jauschalley) [inline src/pages/Project.jsx:36]: Praise: This has nothing to do with this PR but every time I see the Request Service being used I am reminded of how much effort you put into cleaning up this part of the codebase.

File: src/molecules/ProjectHeader.jsx [hunk 4]
```diff
@@ -9,7 +9,7 @@ export default function ProjectHeader({
   title,
 }) {
   const projectHomeUrl = `/projects/${id}`;
-  const projectSettingsUrl = `/projects/${id}/settings`;
+  const projectSettingsUrl = projectHomeUrl + "/settings";
```
Comment (TMCamp) [inline src/molecules/ProjectHeader.jsx:12]: Was `projectHomeUrl` going to be used elsewhere?

File: src/templates/ProjectTemplate.jsx [hunk 5]
```diff
@@ -1,43 +1,59 @@
...
-    <div>
+    <div className="project-page">
       <ProjectHeader
         id={id}
         acronym={acronym}
         impactLevel={impact_level}
         title={title}
       />
-      <ControlsCompleted />
-      {/*@TODO: tooltip is not dispaying correctly <Tooltip className="blueprint-tooltip" position="right" label="System components are group(s) of control narratives that help you create your System Security Plan faster. Components align with your system's architecture, policies and procedures." /> */}
```
Comment (jauschalley) [inline src/templates/ProjectTemplate.jsx:19]: I'm so glad to see these gone.

**Review by jauschalley — APPROVED** [review]

**Review by TMCamp — APPROVED** [review]

File: src/molecules/ProjectHeader.jsx (hunk 4 above)
Comment (jauschalley) [reply src/molecules/ProjectHeader.jsx:12]: It's also used on line 20 of the same file.

**Pull request merged**